### DISCUSSION TITLE
Translations

### DISCUSSION THREAD

**Discussion by Navid200** [NightscoutFoundation/xDrip#3651] [General]
I have always wondered about translations.  I always wonder how a project lead can be confident that no translator has accidentally or unintentionally caused a mistake.

A repository has a lead who has to verify every submission before it is merged.  
A lead knows exactly what field every developer has been active in and which areas the different developers are good at.
So, the lead can ask a developer: "tell me what you think of this PR" before merging or rejecting it.
But, does a lead do the same for every word translation?  I don't see how that would be practical.

I don't want to antagonize anyone here.  I would like to understand how any project lead, not just ours, deals with translations.

I have made submissions for a new feature to be added.  And the reason it was rejected was a single word.  I believe it should be like that.  I'm not questioning that.
But, how can the lead apply the same level of scrutiny if what comes out in the hands of a user in other languages is something that has no idea about?  His scrutiny about the used words only apply to English.  

Diabetes does not spare someone just because they don't speak English.  Everyone can have diabetes.  I don't want to deny anyone the use of xDrip just because they don't speak English.
But, when I was a kid, I didn't speak English.  And I could use a TV even though all the knobs had English descriptions only.  
Of course there is no comparison.  A TV has brightness and contract and volume and few others.  xDrip has so many different settings.   But, I cannot stop wondering is this the right way to do this.

I don't have a solution.  I am just wondering.  How do other projects deal with this?

I have submitted improvements to English strings.  I was asked "are you sure, because if you are not, you will be wasting the time of many volunteers who need to translate these".
I couldn't agree more that I should not change something unnecessarily because it could cause unnecessary work.
But, at the same time, I cannot help wondering isn't this putting the cart before the horse?  

I am sharing my thoughts.  I hope this does not cause any hard feelings.  I want what is best for the project and the volunteers who maintain it. 

If something is wrong in English xDrip, I hear about it.
As an example, there are so many complaints about why there are so many settings in xDrip that are named filter when in fact there is no filtering done for G6 or G7 real data as opposed to the history.  And I intend to fix the definitions in English.  
But, what happens if something is wrong in a different language only because of an incorrect translation?  I don't see how I would hear about that.  How am I supposed to ensure that there are no errors in the definition for different functions in other languages?

**Comment by staticx** [comment]
Thanks Navid for all your hard work. 
Is there a way to help with translation of xDrip to other languages than English? (e.g using crowdin or transifex)

**Reply by Navid200** [reply]
xDrip is already an active project on Crowdin.  xDrip is already translated to many languages.
There are translators who update the strings as new strings are added to xDrip.

**Comment by muro-dot** [comment]
Hello. I am a native Korean speaker. I am a diabetes patient who discovered xdrip a year ago and has been using it ever since. Having seen the already translated app and the progress on Crowdin, I fully empathize with Navid200's concerns.
First, regarding the work I've done as a volunteer: I translated the remaining 85% of untranslated strings using AI, and I've also submitted several requests to correct previous mistranslations. 
While xdrip is perfectly usable in its current state, the real issue lies with the enormous influx of young new patients. As is common everywhere, parents in Korea heavily rely on xdrip for monitoring their children's diabetes. Some mistranslated strings significantly increase the learning curve for first-time users. Therefore, I believe the numerous pending correction requests on Crowdin should be applied more quickly than they are now. 
Because the probability that a requested revision is incorrect compared to the original translation is now extremely low. The era of looking up dictionaries and translating based on personal judgment is over; AI suggests appropriate vocabulary and provides explanations. My submitted translations were processed by AI initially, but only about 5% had awkward phrasing during the secondary review process. 
Please don't agonize over this alone, Navid200. Trust the requests from volunteers, approve them, and move forward. After that, our users' feedback and the volunteers' additional efforts will take over. Additionally, I hope Korean translations can be approved more quickly. Thank you always for your efforts.
